app/javascript/components/visualization/Heatmap.js

```diff
@@ -98,17 +107,22 @@ export default function Heatmap({
 
   return (
     <div className="plot">
+      { ErrorComponent }
       { cluster &&
         <div id={graphId} className="heatmap-graph" style={{ minWidth: '80vw' }}></div> }
       { !cluster && <FontAwesomeIcon icon={faDna} className="gene-load-spinner"/> }
     </div>
   )
 }
 
+
+const SafeHeatmap = withErrorBoundary(Heatmap)
+export default SafeHeatmap
```

Comment:
Interesting - I'd love a dive on `withErrorBoundary` at some point to better understand how it works.